src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -18,7 +21,8 @@ public EqualityArgumentConstraint(object? expectedValue)
 
         public bool IsValid(object? argument)
         {
-            return object.Equals(this.ExpectedValue, argument);
+            var argumentEqualityComparer = ServiceLocator.Resolve<ArgumentEqualityComparer>();
+            return argumentEqualityComparer.AreEqual(this.ExpectedValue, argument, this.parameterType);

```

Comment:
What happens if `AreEqual` throws? I'd at least like to see a test. And maybe we should exhort custom-equalitycomparer-writers to ensure that `AreEqual` doesn't. And I guess `CanCompare`.

Reply:
And `Priority`. Okay. Now I'm going over the top.

Reply:
Ah, yes, need to consider this.

Reply:
I didn't notice a change that addressed it, so I mutated one of your tests to throw a `new Exception("BLAIR")` (so vain) and got

```
[xUnit.net 00:00:01.46]     FakeItEasy.Specs.ArgumentEqualityComparerSpecs.TwoCustomArgumentEqualityComparers() [04] And a call to the fake is made with a distinct but identical instance according to the higher-priority comparer [FAIL]
[xUnit.net 00:00:01.47]     FakeItEasy.Specs.ArgumentEqualityComparerSpecs.TwoCustomArgumentEqualityComparers() [05] Then it should return the configured value [SKIP]
  Failed FakeItEasy.Specs.ArgumentEqualityComparerSpecs.TwoCustomArgumentEqualityComparers() [04] And a call to the fake is made with a distinct but identical instance according to the higher-priority comparer [< 1 ms]
  Error Message:
   System.Exception : BLAIR
  Stack Trace:
     at FakeItEasy.Specs.ArgumentEqualityComparerSpecs.YComparer.AreEqual(ClassWithTwoEligibleArgumentEqualityComparers expectedValue, ClassWithTwoEligibleArgumentEqualityComparers argumentValue) in C:\Dev\FakeItEasy\tests\FakeItEasy.Specs\ArgumentEqualityComparerSpecs.cs:line 95
   at FakeItEasy.ArgumentEqualityComparer`1.AreEqual(Object expectedValue, Object argumentValue) in C:\Dev\FakeItEasy\src\FakeItEasy\ArgumentEqualityComparer.cs:line 32
   at FakeItEasy.Core.ArgumentEqualityComparer.AreEqual(Object expectedValue, Object argumentValue, Type parameterType) in C:\Dev\FakeItEasy\src\FakeItEasy\Core\ArgumentEqualityComparer.cs:line 27
   at FakeItEasy.Expressions.ArgumentConstraints.EqualityArgumentConstraint.IsValid(Object argument) in C:\Dev\FakeItEasy\src\FakeItEasy\Expressions\ArgumentConstraints\EqualityArgumentConstraint.cs:line 24
   at FakeItEasy.Expressions.ExpressionCallMatcher.ArgumentsMatchesArgumentConstraints(ArgumentCollection argumentCollection) in C:\Dev\FakeItEasy\src\FakeItEasy\Expressions\ExpressionCallMatcher.cs:line 115
```

Which isn't terrible, as it does show the source of the error, but I wonder if we shouldn't wrap in a custom exception and indicate that we think this is a configuration error?

Reply:
Ah, yes, forgot to address this. Will do.